ccl/importccl: TestImportIntoCSV failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/65055d6c16bf9386d8c4f4f9cd23e0a848814dc9

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1411253&tab=buildLog

```

I190730 06:30:37.369627 97023 sql/event_log.go:130  [n1,client=127.0.0.1:39950,user=root] Event: "create_table", target: 97, info: {TableName:targetcols.public.target2 Statement:CREATE TABLE target2 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190730 06:30:37.488017 128652 storage/replica_command.go:282  [n1,split,s1,r59/1:/{Table/96/1-Max}] initiating a split of this range at key /Table/97 [r60] (zone config)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3ef8b19]

goroutine 128643 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc002c1a960, 0x5b39460, 0xc004601080)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x48ba360, 0x805b5e0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc004601040, 0x5b39520, 0xc00d63cf90, 0xc00ae881b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc00ae881b0, 0x691031716010001, 0xc0055abd48, 0x7, 0xc0027a4400, 0xc00450d7c0, 0xc00b609320, 0x3f, 0xc001604850)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5b39520, 0xc00d63cf90, 0xc00ae881b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5b39520, 0xc00d63cf90, 0xc00ae881b0, 0x4a2fdadc8ce33b8b, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:627 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc00ae881b0, 0x5b39520, 0xc00d63cf90, 0xc001604840, 0x1, 0xc00ae881b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc002cb8e80, 0x5b39520, 0xc00d63cf90, 0xc0081c2ea0, 0x30, 0x4ac14a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:626 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc00aa92d20, 0x5b39520, 0xc00d63cf90, 0xc0081c2ea0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc00aa92d20, 0x5b39520, 0xc00d63cf90, 0xc003eaab40, 0xd7860e, 0xc007a19cf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc00aa92d20, 0x5b39520, 0xc00d63cf90, 0x5ac42a0, 0xc00b912c80, 0xc00819ab00, 0x5ac42a0, 0xc00b912c80)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5b39520, 0xc00d63cf90)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc002c1a960, 0x5b39460, 0xc004601080, 0xc002de7560, 0x1e, 0x0, 0x0, 0xc002a8c4b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000580800, 0x4cc89ca, 0x11, 0x4e14ad0, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc000580800)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc000580800, 0xc0007edd68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc00029fe40, 0x8083300, 0x1c, 0x1c, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc00054cc80, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
github.com/cockroachdb/cockroach/pkg/ccl/importccl.TestMain(0xc00054cc80)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/main_test.go:31 +0x180
main.main()
	_testmain.go:108 +0x223

```

SHA: https://github.com/cockroachdb/cockroach/commits/92fef12128c997233d985d1c19e11faac005073f

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1413454&tab=buildLog

```

I190731 07:21:06.599044 97777 sql/event_log.go:130  [n1,client=127.0.0.1:57274,user=root] Event: "create_table", target: 97, info: {TableName:targetcols.public.target2 Statement:CREATE TABLE target2 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190731 07:21:06.602308 128589 storage/replica_command.go:282  [n1,split,s1,r59/1:/{Table/96/1-Max}] initiating a split of this range at key /Table/97 [r60] (zone config)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3f1fc39]

goroutine 128940 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc002b8d0e0, 0x5b6de40, 0xc0031d0e40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x48e5f00, 0x80ab680)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc0031d0e00, 0x5b6df00, 0xc007624150, 0xc009c435f0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc009c435f0, 0x6920d9c2caa0001, 0xc0079043e8, 0x7, 0xc00111ed80, 0xc003a2fac0, 0xc003d53ec0, 0x3f, 0xc003a61580)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5b6df00, 0xc007624150, 0xc009c435f0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5b6df00, 0xc007624150, 0xc009c435f0, 0xf7ff887bbd1568ba, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:628 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc009c435f0, 0x5b6df00, 0xc007624150, 0xc003a61570, 0x1, 0xc009c435f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc0011a0400, 0x5b6df00, 0xc007624150, 0xc000eb04e0, 0x30, 0x4af0220)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:627 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc000f6e460, 0x5b6df00, 0xc007624150, 0xc000eb04e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc000f6e460, 0x5b6df00, 0xc007624150, 0xc007fc8c80, 0xd7860e, 0xc00b991cf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc000f6e460, 0x5b6df00, 0xc007624150, 0x5af85c0, 0xc0026be180, 0xc0028d9560, 0x5af85c0, 0xc0026be180)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5b6df00, 0xc007624150)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc002b8d0e0, 0x5b6de40, 0xc0031d0e40, 0xc00b965ae0, 0x1e, 0x0, 0x0, 0xc001f7e6e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0001a7a00, 0x4cf886c, 0x11, 0x4e45848, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc0001a7a00)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc0001a7a00, 0xc000811d68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc0002f5700, 0x80d3900, 0x1c, 0x1c, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc000124300, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
github.com/cockroachdb/cockroach/pkg/ccl/importccl.TestMain(0xc000124300)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/main_test.go:31 +0x180
main.main()
	_testmain.go:108 +0x223

```

SHA: https://github.com/cockroachdb/cockroach/commits/da56c792e968574b8f1d9ef3fdb45d56a530221a

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1415626&tab=buildLog

```

I190801 08:35:03.301418 97104 sql/event_log.go:130  [n1,client=127.0.0.1:55778,user=root] Event: "create_table", target: 97, info: {TableName:targetcols.public.target2 Statement:CREATE TABLE target2 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190801 08:35:03.306508 128096 storage/replica_command.go:282  [n1,split,s1,r59/1:/{Table/96/1-Max}] initiating a split of this range at key /Table/97 [r60] (zone config)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3f24949]

goroutine 128512 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc005e97cc0, 0x5b76520, 0xc005dc44c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x48eae80, 0x80b5680)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc005dc42c0, 0x5b765e0, 0xc002954e40, 0xc005be53b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc005be53b0, 0x6931c5312fb0001, 0xc0019d4e19, 0x7, 0xc005fbdf00, 0xc000f67c80, 0xc003601a70, 0x3f, 0xc001e4ae10)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5b765e0, 0xc002954e40, 0xc005be53b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5b765e0, 0xc002954e40, 0xc005be53b0, 0x472b53a6558656b8, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:628 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc005be53b0, 0x5b765e0, 0xc002954e40, 0xc001e4ae00, 0x1, 0xc005be53b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc00289c880, 0x5b765e0, 0xc002954e40, 0xc00690eae0, 0x30, 0x4af5180)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:627 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc007ef57a0, 0x5b765e0, 0xc002954e40, 0xc00690eae0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc007ef57a0, 0x5b765e0, 0xc002954e40, 0xc0024fabc0, 0xd7860e, 0xc008033cf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc007ef57a0, 0x5b765e0, 0xc002954e40, 0x5b00be0, 0xc00386f200, 0xc00680f500, 0x5b00be0, 0xc00386f200)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5b765e0, 0xc002954e40)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc005e97cc0, 0x5b76520, 0xc005dc44c0, 0xc009ccd8a0, 0x1e, 0x0, 0x0, 0xc008941f40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00072c100, 0x4cfdab7, 0x11, 0x4e4b608, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc00072c100)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc00072c100, 0xc0007add68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc0002de960, 0x80dd900, 0x1c, 0x1c, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc000722200, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
github.com/cockroachdb/cockroach/pkg/ccl/importccl.TestMain(0xc000722200)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/main_test.go:31 +0x180
main.main()
	_testmain.go:108 +0x223

```

SHA: https://github.com/cockroachdb/cockroach/commits/175c5ada040fd0cbbf178636b1c551d5c2229ec4

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1417689&tab=buildLog

```

I190802 05:48:08.349895 98186 sql/event_log.go:130  [n1,client=127.0.0.1:37788,user=root] Event: "create_database", target: 88, info: {DatabaseName:csv18 Statement:CREATE DATABASE csv18 User:root}
I190802 05:48:08.417117 98186 sql/event_log.go:130  [n1,client=127.0.0.1:37788,user=root] Event: "create_table", target: 89, info: {TableName:csv18.public.t Statement:CREATE TABLE t (a INT8, b STRING) User:root}
I190802 05:48:08.481217 121583 storage/replica_command.go:283  [n1,split,s1,r51/1:/{Table/87-Max}] initiating a split of this range at key /Table/89 [r52] (zone config)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3f2bbb9]

goroutine 121533 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0042248c0, 0x5b84b40, 0xc007801fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x48f34c0, 0x80c8680)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc007801f80, 0x5b84c00, 0xc000e761b0, 0xc006aa2480, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc006aa2480, 0x693fff86fe38001, 0xc0035b81b8, 0x7, 0xc0031bb280, 0xc000dbfa40, 0xc003058bd0, 0x3f, 0xc002f6a900)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5b84c00, 0xc000e761b0, 0xc006aa2480, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5b84c00, 0xc000e761b0, 0xc006aa2480, 0x3670074153006188, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:628 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc006aa2480, 0x5b84c00, 0xc000e761b0, 0xc002f6a8f0, 0x1, 0xc006aa2480)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc001a35900, 0x5b84c00, 0xc000e761b0, 0xc002a4e0f0, 0x30, 0x4afdb20)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:627 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc001ea0d20, 0x5b84c00, 0xc000e761b0, 0xc002a4e0f0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc001ea0d20, 0x5b84c00, 0xc000e761b0, 0xc002c57e00, 0xd7860e, 0xc00814bcf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc001ea0d20, 0x5b84c00, 0xc000e761b0, 0x5b0ef80, 0xc0033ee9e0, 0xc00379d620, 0x5b0ef80, 0xc0033ee9e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5b84c00, 0xc000e761b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc0042248c0, 0x5b84b40, 0xc007801fc0, 0xc007e00900, 0x1e, 0x0, 0x0, 0xc007ec4f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000506100, 0x4d06b07, 0x11, 0x4e549e0, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc000506100)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc000506100, 0xc0007ebd68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc00030e0e0, 0x80f0920, 0x1c, 0x1c, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc000584580, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
```

SHA: https://github.com/cockroachdb/cockroach/commits/4bc081bee5a52c4b8116910025c7c04519872e99

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1420192&tab=buildLog

```
--- FAIL: TestImportIntoCSV (49.16s)
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr
I190803 06:22:19.109577 97571 sql/event_log.go:130  [n1,client=127.0.0.1:57808,user=root] Event: "create_table", target: 95, info: {TableName:targetcols.public.target0 Statement:CREATE TABLE target0 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190803 06:22:19.217523 126228 storage/replica_command.go:283  [n1,split,s1,r56/1:/{Table/94-Max}] initiating a split of this range at key /Table/95 [r57] (zone config)
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr (0.45s)
            import_stmt_test.go:1579: error executing 'IMPORT INTO target0 (a) CSV DATA ($1)': pq: another operation is currently operating on the table: unexpected value: raw_bytes:"U\306\257>\003\n\206\003\n\007target0\030_ ^(\001:\n\010\307\310\315\350\216\256\325\333\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001b\020\002\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001c\020\003\032\n\010\007\020\000\030\0000\000P\031 \0010\000B\025\n\001d\020\004\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001e\020\005\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001f\020\006\032\n\010\007\020\000\030\0000\000P\031 \0010\000B)\n\005rowid\020\007\032\n\010\001\020@\030\0000\000P\024 \000*\016unique_rowid()0\001H\010R3\n\007primary\020\001\030\001\"\005rowid0\007@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\0014\n\007primary\020\000\032\001a\032\001b\032\001c\032\001d\032\001e\032\001f\032\005rowid \001 \002 \003 \004 \005 \006 \007(\000\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\307\310\315\350\216\256\325\333\025" timestamp:<wall_time:1564813339036705863 > 
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#01
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#01 (0.01s)
            import_stmt_test.go:1587: error executing 'CREATE TABLE target0 (a INT8,
                			b INT8,
                			c STRING,
                			d INT8,
                			e INT8,
                			f STRING)': pq: relation "target0" already exists
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#02
I190803 06:22:19.654096 96890 server/status/runtime.go:498  [n1] runtime stats: 1.6 GiB RSS, 843 goroutines, 118 MiB/111 MiB/270 MiB GO alloc/idle/total, 119 MiB/167 MiB CGO alloc/total, 4411.2 CGO/sec, 354.3/39.9 %(u/s)time, 0.1 %gc (11x), 2.9 MiB/2.9 MiB (r/w)net
I190803 06:22:19.666372 97571 sql/sqlbase/structured.go:1372  [n1,client=127.0.0.1:57808,user=root] publish: descID=95 (target0) version=2 mtime=2019-08-03 06:22:19.514911443 +0000 UTC
I190803 06:22:19.818723 97571 sql/event_log.go:130  [n1,client=127.0.0.1:57808,user=root] Event: "drop_database", target: 94, info: {DatabaseName:targetcols Statement:DROP DATABASE targetcols User:root DroppedSchemaObjects:[targetcols.public.target0]}
I190803 06:22:19.879245 97571 sql/sqlbase/structured.go:1372  [n1,client=127.0.0.1:57808,user=root,scExec] publish: descID=95 (target0) version=3 mtime=2019-08-03 06:22:19.873762034 +0000 UTC
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#02 (0.01s)
            import_stmt_test.go:1597: error executing 'CREATE TABLE target0 (a INT8,
                			b INT8,
                			c STRING,
                			d INT8,
                			e INT8,
                			f STRING)': pq: relation "target0" already exists
=== RUN   TestImportIntoCSV/target-cols-with-default-pk
I190803 06:22:19.031652 97571 sql/event_log.go:130  [n1,client=127.0.0.1:57808,user=root] Event: "create_database", target: 94, info: {DatabaseName:targetcols Statement:CREATE DATABASE targetcols User:root}
    --- FAIL: TestImportIntoCSV/target-cols-with-default-pk (1.17s)

```

SHA: https://github.com/cockroachdb/cockroach/commits/3b9a95bd7eb2cfa6d544fe7217852a85ec3b76f4

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1421860&tab=buildLog

```

I190804 05:20:15.380742 96695 sql/event_log.go:130  [n1,client=127.0.0.1:51102,user=root] Event: "create_table", target: 97, info: {TableName:targetcols.public.target2 Statement:CREATE TABLE target2 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190804 05:20:15.381345 94923 server/status/runtime.go:498  [n2] runtime stats: 1.6 GiB RSS, 859 goroutines, 134 MiB/97 MiB/270 MiB GO alloc/idle/total, 120 MiB/162 MiB CGO alloc/total, 3580.5 CGO/sec, 317.4/35.6 %(u/s)time, 0.1 %gc (8x), 4.5 MiB/4.5 MiB (r/w)net
I190804 05:20:15.450557 128411 storage/replica_command.go:283  [n1,split,s1,r59/1:/{Table/96/1-Max}] initiating a split of this range at key /Table/97 [r60] (zone config)
I190804 05:20:15.708717 95123 server/status/runtime.go:498  [n3] runtime stats: 1.7 GiB RSS, 869 goroutines, 154 MiB/78 MiB/270 MiB GO alloc/idle/total, 121 MiB/166 MiB CGO alloc/total, 3553.6 CGO/sec, 317.8/36.4 %(u/s)time, 0.1 %gc (8x), 4.5 MiB/4.5 MiB (r/w)net
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3f3b799]

goroutine 128693 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc001785680, 0x5ba30c0, 0xc0033e6e80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x4902e00, 0x80f1680)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc0033e6e40, 0x5ba3180, 0xc006ddc120, 0xc00b134870, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc00b134870, 0x695fdf7d21d8001, 0xc003fbec19, 0x7, 0xc001edfa80, 0xc0049b6f00, 0xc0018c1dd0, 0x3f, 0xc0024509c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5ba3180, 0xc006ddc120, 0xc00b134870, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5ba3180, 0xc006ddc120, 0xc00b134870, 0x80ad828a1a5d6db2, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:633 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc00b134870, 0x5ba3180, 0xc006ddc120, 0xc0024509a0, 0x1, 0xc00b134870)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc004593300, 0x5ba3180, 0xc006ddc120, 0xc0083e03c0, 0x30, 0x4b0dfc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:632 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc008d31880, 0x5ba3180, 0xc006ddc120, 0xc0083e03c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc008d31880, 0x5ba3180, 0xc006ddc120, 0xc0014a88c0, 0xd7862e, 0xc006abfcf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc008d31880, 0x5ba3180, 0xc006ddc120, 0x5b2d380, 0xc007985b00, 0xc007bc5d00, 0x5b2d380, 0xc007985b00)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5ba3180, 0xc006ddc120)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc001785680, 0x5ba30c0, 0xc0033e6e80, 0xc000448200, 0x1e, 0x0, 0x0, 0xc006670be0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000252100, 0x4d178f4, 0x11, 0x4e65b70, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc000252100)
```

SHA: https://github.com/cockroachdb/cockroach/commits/3db89b230b0c41e399354cbeb78c1e82c8e30004

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1424368&tab=buildLog

```

I190806 07:09:40.307448 97297 sql/event_log.go:130  [n1,client=127.0.0.1:33698,user=root] Event: "create_table", target: 96, info: {TableName:targetcols.public.target1 Statement:CREATE TABLE target1 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3f38659]

goroutine 127422 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc004ecec80, 0x5b9ee20, 0xc004782740)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:181 +0x155
panic(0x48feea0, 0x80eb680)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).OnFailOrCancel(0xc004782700, 0x5b9eee0, 0xc001861260, 0xc007908900, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:928 +0x2e9
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed.func1(0xc007908900, 0x698148465840001, 0xc004ea0ca8, 0x7, 0xc003414f80, 0xc00352e680, 0xc001dae120, 0x3f, 0xc00204ffd0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:345 +0xd6
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update.func1(0x5b9eee0, 0xc001861260, 0xc007908900, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:133 +0x575
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x5b9eee0, 0xc001861260, 0xc007908900, 0x70016c641c838398, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:633 +0x5f
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc007908900, 0x5b9eee0, 0xc001861260, 0xc00204ffb0, 0x1, 0xc007908900)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:686 +0xea
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc002143e80, 0x5b9eee0, 0xc001861260, 0xc00bcad6e0, 0x30, 0x4b0a440)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:632 +0x162
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).runInTxn(0xc000a721c0, 0x5b9eee0, 0xc001861260, 0xc00bcad6e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:438 +0x14d
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Update(0xc000a721c0, 0x5b9eee0, 0xc001861260, 0xc00234eec0, 0xd7862e, 0xc006bdbcf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/update.go:104 +0x22e
github.com/cockroachdb/cockroach/pkg/jobs.(*Job).Failed(0xc000a721c0, 0x5b9eee0, 0xc001861260, 0x5b28ee0, 0xc00039a520, 0xc0048afb00, 0x5b28ee0, 0xc00039a520)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:340 +0x144
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x5b9eee0, 0xc001861260)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:584 +0x6c7
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc004ecec80, 0x5b9ee20, 0xc004782740, 0xc001b21e20, 0x1e, 0x0, 0x0, 0xc00437af50)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:321 +0xf4
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:316 +0x14b

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0001d3200, 0x4d13cef, 0x11, 0x4e62150, 0x1)
	/usr/local/go/src/testing/testing.go:917 +0x693
testing.runTests.func1(0xc0001d3200)
	/usr/local/go/src/testing/testing.go:1157 +0xa9
testing.tRunner(0xc0001d3200, 0xc000819d68)
	/usr/local/go/src/testing/testing.go:865 +0x164
testing.runTests(0xc000278de0, 0x8113900, 0x1c, 0x1c, 0x0)
	/usr/local/go/src/testing/testing.go:1155 +0x524
testing.(*M).Run(0xc000242300, 0x0)
	/usr/local/go/src/testing/testing.go:1072 +0x2ec
github.com/cockroachdb/cockroach/pkg/ccl/importccl.TestMain(0xc000242300)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/main_test.go:31 +0x180
main.main()
	_testmain.go:108 +0x223

goroutine 36 [syscall, 1 minutes]:
os/signal.signal_recv(0x759f21)
	/usr/local/go/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x30
```

SHA: https://github.com/cockroachdb/cockroach/commits/19ad2eaaa2b01d8ba0e1d89b1d6e5eb0dea35630

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1429424&tab=buildLog

```
I190808 05:25:02.255792 96399 server/status/runtime.go:498  [n2] runtime stats: 2.1 GiB RSS, 943 goroutines, 150 MiB/80 MiB/271 MiB GO alloc/idle/total, 177 MiB/218 MiB CGO alloc/total, 1795.4 CGO/sec, 312.6/36.6 %(u/s)time, 0.3 %gc (6x), 3.0 MiB/3.0 MiB (r/w)net
W190808 05:25:02.362217 171273 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.386110 95541 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.386675 95541 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.388649 96325 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.389294 96325 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.389990 96678 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:02.390610 96678 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190808 05:25:02.680099 96726 server/status/runtime.go:498  [n3] runtime stats: 2.1 GiB RSS, 929 goroutines, 178 MiB/55 MiB/271 MiB GO alloc/idle/total, 177 MiB/219 MiB CGO alloc/total, 1798.9 CGO/sec, 314.6/36.5 %(u/s)time, 0.3 %gc (6x), 3.0 MiB/3.0 MiB (r/w)net
W190808 05:25:03.736184 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.749781 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.760641 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.767757 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.773951 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.781897 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.790573 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.797134 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.805989 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:25:03.817408 172426 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
    --- FAIL: TestImportIntoCSV/direct-import-into-key-collision (3.22s)
        import_stmt_test.go:1923: expected error 'ingested key collides with an existing one: /Table/121/1/0/0', got: pq: addsstable [/Table/122/1/0/0,/Table/122/1/35/0/NULL): checking for key collisions: ingested key collides with an existing one: /Table/122/1/0/0

```

SHA: https://github.com/cockroachdb/cockroach/commits/19ad2eaaa2b01d8ba0e1d89b1d6e5eb0dea35630

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1429423&tab=buildLog

```
W190808 05:29:57.471918 580375 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.475523 580283 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.475757 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.479476 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.480165 580285 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.483874 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.484476 580267 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.487491 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.488988 580334 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.491275 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.492788 580336 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.494716 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.496058 580269 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.499229 580271 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.501955 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.503547 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.503712 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.507165 580235 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.508167 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190808 05:29:57.508447 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
    --- FAIL: TestImportIntoCSV/import-into-same-file-diff-imports (0.24s)
        import_stmt_test.go:1866: expected error 'ingested key collides with an existing one: /Table/115/1/0/0', got: pq: addsstable [/Table/115/1/765/0,/Table/115/1/1000): checking for key collisions: ingested key collides with an existing one: /Table/115/1/765/0

```

SHA: https://github.com/cockroachdb/cockroach/commits/ca494e6d787a2fb516e3e1036a46917fbe4b8458

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1432367&tab=buildLog

```
I190809 06:08:35.054012 97411 sql/event_log.go:130  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I190809 06:08:35.174335 97586 sql/event_log.go:130  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.1-7 User:root}
I190809 06:08:35.251851 97716 sql/event_log.go:130  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I190809 06:08:35.393826 97927 sql/event_log.go:130  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:cbf75619-9d29-4f83-b789-327485e0a15d User:root}
I190809 06:08:35.461161 98002 sql/event_log.go:130  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I190809 06:08:35.499549 98011 sql/event_log.go:130  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I190809 06:08:35.625248 95293 server/server.go:1548  [n1] done ensuring all necessary migrations have run
I190809 06:08:35.630019 95293 server/server.go:1786  [n1] serving sql connections
I190809 06:08:35.653406 97881 server/server_update.go:68  [n1] no need to upgrade, cluster already at the newest version
I190809 06:08:35.663493 97883 sql/event_log.go:130  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:33315 Attrs: Locality: ServerVersion:19.1-7 BuildTag:v19.2.0-alpha.20190606-1413-gca494e6 StartedAt:1565330914743863128 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:35021} ClusterID:ab449350-0aeb-49e3-8041-993c9f47ef4a StartedAt:1565330914743863128 LastUp:1565330914743863128}
I190809 06:08:35.715350 98110 sql/event_log.go:130  [n1,client=127.0.0.1:51740,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.import.batch_size Value:10KB User:root}
--- FAIL: TestImportIntoCSV (77.69s)
=== RUN   TestImportIntoCSV/import-into-no-decompress-gzip
I190809 06:09:03.845825 98110 sql/event_log.go:130  [n1,client=127.0.0.1:51740,user=root] Event: "create_database", target: 86, info: {DatabaseName:csv17 Statement:CREATE DATABASE csv17 User:root}
I190809 06:09:03.955939 98110 sql/event_log.go:130  [n1,client=127.0.0.1:51740,user=root] Event: "create_table", target: 87, info: {TableName:csv17.public.t Statement:CREATE TABLE t (a INT8, b STRING) User:root}
I190809 06:09:03.972988 120172 storage/replica_command.go:283  [n1,split,s1,r50/1:/{Table/85-Max}] initiating a split of this range at key /Table/87 [r51] (zone config)
I190809 06:09:04.320854 95463 storage/queue.go:518  [n1,s1,r11/1:/Table/1{5-6}] rate limited in MaybeAdd (raftlog): context canceled
    --- FAIL: TestImportIntoCSV/import-into-no-decompress-gzip (0.76s)
        import_stmt_test.go:1431: expected error 'expected 2 fields, got', got: pq: another operation is currently operating on the table: unexpected value: raw_bytes:"u\341\312\251\003\n\220\002\n\001t\030W V(\001:\n\010\323\331\370\201\263\206\313\334\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001b\020\002\032\n\010\007\020\000\030\0000\000P\031 \0010\000B)\n\005rowid\020\003\032\n\010\001\020@\030\0000\000P\024 \000*\016unique_rowid()0\001H\004R3\n\007primary\020\001\030\001\"\005rowid0\003@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001 \n\007primary\020\000\032\001a\032\001b\032\005rowid \001 \002 \003(\000\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\323\331\370\201\263\206\313\334\025" timestamp:<wall_time:1565330943852293331 > 

```

SHA: https://github.com/cockroachdb/cockroach/commits/98d6832e9f9edb7e554aaa90d9d4296bb00af16e

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1433743&tab=buildLog

```
I190810 05:24:40.579825 116214 storage/replica_raft.go:290  [n1,s1,r1/1:/{Min-System/NodeL…}] proposing ADD_REPLICA((n2,s2):2): updated=(n1,s1):1,(n2,s2):2 next=3
I190810 05:24:40.593058 116214 storage/store_snapshot.go:984  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] sending PREEMPTIVE snapshot 370f09aa at applied index 102
I190810 05:24:40.594170 116214 storage/store_snapshot.go:1027  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] streamed snapshot to (n3,s3):?: kv pairs: 67, log entries: 0, rate-limit: 8.0 MiB/sec, 0.00s
I190810 05:24:40.598216 118209 storage/replica_raftstorage.go:811  [n3,s3,r1/?:{-}] applying PREEMPTIVE snapshot [id=370f09aa index=102]
W190810 05:24:40.598787 118209 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190810 05:24:40.599446 118209 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190810 05:24:40.599732 79714 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190810 05:24:40.600060 79714 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190810 05:24:40.600505 118209 storage/replica_raftstorage.go:836  [n3,s3,r1/?:/{Min-System/NodeL…}] applied PREEMPTIVE snapshot [total=2ms unreplicatedState=0ms ingestion=4@1ms id=370f09aa index=102]
I190810 05:24:40.603281 116214 storage/replica_command.go:1188  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] change replicas (ADD_REPLICA (n3,s3):3): existing descriptor r1:/{Min-System/NodeLiveness} [(n1,s1):1, (n2,s2):2, next=3, gen=1]
I190810 05:24:40.623282 116214 storage/replica_raft.go:290  [n1,s1,r1/1:/{Min-System/NodeL…}] proposing ADD_REPLICA((n3,s3):3): updated=(n1,s1):1,(n2,s2):2,(n3,s3):3 next=4
I190810 05:24:40.653888 116214 testutils/testcluster/testcluster.go:655  WaitForFullReplication took: 1.910131266s
I190810 05:24:40.750893 118132 sql/event_log.go:130  [n1,client=127.0.0.1:41868,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.import.batch_size Value:10KB User:root}
--- FAIL: TestImportIntoCSV (48.47s)
=== RUN   TestImportIntoCSV/import-into-dups-in-sst
I190810 05:25:19.690239 118132 sql/event_log.go:130  [n1,client=127.0.0.1:41868,user=root] Event: "create_database", target: 116, info: {DatabaseName:sameimport Statement:CREATE DATABASE sameimport User:root}
I190810 05:25:19.766181 118132 sql/event_log.go:130  [n1,client=127.0.0.1:41868,user=root] Event: "create_table", target: 117, info: {TableName:sameimport.public.t Statement:CREATE TABLE t (a INT8 PRIMARY KEY, b STRING) User:root}
I190810 05:25:19.927600 174149 storage/replica_command.go:283  [n1,split,s1,r114/1:/{Table/115/1-Max}] initiating a split of this range at key /Table/117 [r115] (zone config)
    --- FAIL: TestImportIntoCSV/import-into-dups-in-sst (0.44s)
        import_stmt_test.go:1817: error executing 'IMPORT INTO t (a, b) CSV DATA ('nodelocal:///dup-key-same-value-race') WITH experimental_direct_ingestion': pq: another operation is currently operating on the table: unexpected value: raw_bytes:"\200%\246\026\003\n\330\001\n\001t\030u t(\001:\n\010\333\323\334\255\314\214\336\334\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0000\000B\025\n\001b\020\002\032\n\010\007\020\000\030\0000\000P\031 \0010\000H\003R/\n\007primary\020\001\030\001\"\001a0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\027\n\007primary\020\000\032\001a\032\001b \001 \002(\002\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\333\323\334\255\314\214\336\334\025" timestamp:<wall_time:1565414719697136091 > 

```

SHA: https://github.com/cockroachdb/cockroach/commits/14f9c378040b933900a84d11045f1750440b2c71

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1436253&tab=buildLog

```
I190813 05:23:52.949679 116949 storage/replica_raftstorage.go:811  [n2,s2,r5/?:{-}] applying PREEMPTIVE snapshot [id=355af0f6 index=15]
W190813 05:23:52.950317 116949 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190813 05:23:52.951127 116949 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190813 05:23:52.951461 92422 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190813 05:23:52.951832 92422 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190813 05:23:52.952480 116949 storage/replica_raftstorage.go:836  [n2,s2,r5/?:/{Systemtse-Table/System…}] applied PREEMPTIVE snapshot [total=3ms unreplicatedState=0ms ingestion=4@1ms id=355af0f6 index=15]
I190813 05:23:52.957153 114683 storage/replica_command.go:1188  [n1,replicate,s1,r5/1:/{Systemtse-Table/System…}] change replicas (ADD_REPLICA (n2,s2):3): existing descriptor r5:/{Systemtse-Table/SystemConfigSpan/Start} [(n1,s1):1, (n3,s3):2, next=3, gen=1]
I190813 05:23:52.970548 116083 gossip/gossip.go:1531  [n3] node has connected to cluster via gossip
I190813 05:23:52.972226 116083 storage/stores.go:259  [n3] wrote 2 node addresses to persistent storage
I190813 05:23:52.978165 114683 storage/replica_raft.go:290  [n1,s1,r5/1:/{Systemtse-Table/System…}] proposing ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I190813 05:23:53.013396 114683 testutils/testcluster/testcluster.go:655  WaitForFullReplication took: 2.165969788s
I190813 05:23:53.111895 117004 sql/event_log.go:130  [n1,client=127.0.0.1:51662,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.import.batch_size Value:10KB User:root}
--- FAIL: TestImportIntoCSV (49.82s)
=== RUN   TestImportIntoCSV/multiple-file-import-into-without-pk
I190813 05:24:28.575125 117004 sql/event_log.go:130  [n1,client=127.0.0.1:51662,user=root] Event: "create_database", target: 112, info: {DatabaseName:nokeycollision Statement:CREATE DATABASE nokeycollision User:root}
I190813 05:24:28.685571 117004 sql/event_log.go:130  [n1,client=127.0.0.1:51662,user=root] Event: "create_table", target: 113, info: {TableName:nokeycollision.public.t Statement:CREATE TABLE t (a INT8, b STRING) User:root}
I190813 05:24:28.834722 167156 storage/replica_command.go:283  [n1,split,s1,r109/1:/{Table/111/1-Max}] initiating a split of this range at key /Table/113 [r110] (zone config)
    --- FAIL: TestImportIntoCSV/multiple-file-import-into-without-pk (0.55s)
        import_stmt_test.go:1772: error executing 'IMPORT INTO t (a, b) CSV DATA ('nodelocal:///data-0-race', 'nodelocal:///data-0-race')': pq: could not mark job 477113446275710977 as failed: another operation is currently operating on the table: unexpected value: raw_bytes:"6\355\031\324\003\n\220\002\n\001t\030q p(\001:\n\010\241\221\311\226\351\202\231\335\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001b\020\002\032\n\010\007\020\000\030\0000\000P\031 \0010\000B)\n\005rowid\020\003\032\n\010\001\020@\030\0000\000P\024 \000*\016unique_rowid()0\001H\004R3\n\007primary\020\001\030\001\"\005rowid0\003@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001 \n\007primary\020\000\032\001a\032\001b\032\005rowid \001 \002 \003(\000\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\241\221\311\226\351\202\231\335\025" timestamp:<wall_time:1565673868579981473 > : invalid pre-IMPORT time to rollback

```

SHA: https://github.com/cockroachdb/cockroach/commits/bad5e2ea8674f0513f949d3940609b1f4d685581

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1438011&tab=buildLog

```
I190814 07:49:22.605367 115320 storage/replica_raft.go:291  [n1,s1,r15/1:/Table/{19-20}] proposing ADD_REPLICA[(n3,s3):2]: after=[(n1,s1):1 (n3,s3):2] next=3
I190814 07:49:22.610934 115320 storage/replica_command.go:1198  [n1,replicate,s1,r15/1:/Table/{19-20}] change replicas (ADD_REPLICA (n2,s2):3LEARNER): existing descriptor r15:/Table/{19-20} [(n1,s1):1, (n3,s3):2, next=3, gen=2]
I190814 07:49:22.635629 115320 storage/replica_raft.go:291  [n1,s1,r15/1:/Table/{19-20}] proposing ADD_REPLICA[(n2,s2):3LEARNER]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3LEARNER] next=4
I190814 07:49:22.669095 115320 storage/store_snapshot.go:786  [n1,replicate,s1,r15/1:/Table/{19-20}] sending LEARNER snapshot 2770f92c at applied index 19
I190814 07:49:22.670176 115320 storage/store_snapshot.go:829  [n1,replicate,s1,r15/1:/Table/{19-20}] streamed snapshot to (n2,s2):3: kv pairs: 10, log entries: 0, rate-limit: 8.0 MiB/sec, 0.01s
I190814 07:49:22.674772 117718 storage/replica_raftstorage.go:823  [n2,s2,r15/3:{-}] applying LEARNER snapshot at index 19 (id=2770f92c, encoded size=1207, 1 rocksdb batches, 0 log entries)
I190814 07:49:22.676830 117718 storage/replica_raftstorage.go:829  [n2,s2,r15/3:/Table/{19-20}] applied LEARNER snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I190814 07:49:22.685826 115320 storage/replica_command.go:1198  [n1,replicate,s1,r15/1:/Table/{19-20}] change replicas (ADD_REPLICA (n2,s2):3): existing descriptor r15:/Table/{19-20} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I190814 07:49:22.732302 115320 storage/replica_raft.go:291  [n1,s1,r15/1:/Table/{19-20}] proposing ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I190814 07:49:22.773004 115320 testutils/testcluster/testcluster.go:655  WaitForFullReplication took: 3.297523167s
I190814 07:49:22.860356 117722 sql/event_log.go:130  [n1,client=127.0.0.1:48014,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.import.batch_size Value:10KB User:root}
--- FAIL: TestImportIntoCSV (50.90s)
=== RUN   TestImportIntoCSV/import-into-same-file-diff-imports
I190814 07:50:00.015228 117722 sql/event_log.go:130  [n1,client=127.0.0.1:48014,user=root] Event: "create_database", target: 114, info: {DatabaseName:keycollision Statement:CREATE DATABASE keycollision User:root}
I190814 07:50:00.136076 117722 sql/event_log.go:130  [n1,client=127.0.0.1:48014,user=root] Event: "create_table", target: 115, info: {TableName:keycollision.public.t Statement:CREATE TABLE t (a INT8 PRIMARY KEY, b STRING) User:root}
I190814 07:50:00.198723 169135 storage/replica_command.go:283  [n1,split,s1,r111/1:/{Table/113/1-Max}] initiating a split of this range at key /Table/115 [r113] (zone config)
I190814 07:50:00.338835 116501 storage/queue.go:518  [n3,s3,r111/3:/Table/11{3/1-5}] rate limited in MaybeAdd (split): throttled on async limiting semaphore
    --- FAIL: TestImportIntoCSV/import-into-same-file-diff-imports (0.75s)
        import_stmt_test.go:1794: error executing 'IMPORT INTO t (a, b) CSV DATA ('nodelocal:///data-0-race') WITH experimental_direct_ingestion': pq: could not mark job 477425173011136513 as failed: another operation is currently operating on the table: unexpected value: raw_bytes:"0\352e$\003\n\330\001\n\001t\030s r(\001:\n\010\334\272\355\277\301\323\256\335\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0000\000B\025\n\001b\020\002\032\n\010\007\020\000\030\0000\000P\031 \0010\000H\003R/\n\007primary\020\001\030\001\"\001a0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\027\n\007primary\020\000\032\001a\032\001b \001 \002(\002\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\334\272\355\277\301\323\256\335\025" timestamp:<wall_time:1565769000044682588 > : invalid pre-IMPORT time to rollback

```

SHA: https://github.com/cockroachdb/cockroach/commits/02f239d810711786174736328e3360dccf7ef412

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportIntoCSV PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1439812&tab=buildLog

```
I190815 05:25:15.212717 118401 sql/event_log.go:130  [n1,client=127.0.0.1:59046,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.import.batch_size Value:10KB User:root}
--- FAIL: TestImportIntoCSV (55.76s)
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr
I190815 05:25:41.790352 118401 sql/event_log.go:130  [n1,client=127.0.0.1:59046,user=root] Event: "create_table", target: 95, info: {TableName:targetcols.public.target0 Statement:CREATE TABLE target0 (a INT8, b INT8, c STRING, d INT8, e INT8, f STRING) User:root}
I190815 05:25:41.833295 143810 storage/replica_command.go:283  [n1,split,s1,r91/1:/{Table/93/1-Max}] initiating a split of this range at key /Table/95 [r92] (zone config)
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr (0.58s)
            import_stmt_test.go:1567: error executing 'IMPORT INTO target0 (a) CSV DATA ($1)': pq: could not mark job 477679916834979841 as failed: another operation is currently operating on the table: unexpected value: raw_bytes:"\014\010M\010\003\n\206\003\n\007target0\030_ ^(\001:\n\010\331\222\320\321\213\252\300\335\025B\025\n\001a\020\001\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001b\020\002\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001c\020\003\032\n\010\007\020\000\030\0000\000P\031 \0010\000B\025\n\001d\020\004\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001e\020\005\032\n\010\001\020@\030\0000\000P\024 \0010\000B\025\n\001f\020\006\032\n\010\007\020\000\030\0000\000P\031 \0010\000B)\n\005rowid\020\007\032\n\010\001\020@\030\0000\000P\024 \000*\016unique_rowid()0\001H\010R3\n\007primary\020\001\030\001\"\005rowid0\007@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\0014\n\007primary\020\000\032\001a\032\001b\032\001c\032\001d\032\001e\032\001f\032\005rowid \001 \002 \003 \004 \005 \006 \007(\000\270\001\001\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\331\222\320\321\213\252\300\335\025" timestamp:<wall_time:1565846741674363225 > : invalid pre-IMPORT time to rollback
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#01
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#01 (0.02s)
            import_stmt_test.go:1575: error executing 'CREATE TABLE target0 (a INT8,
                			b INT8,
                			c STRING,
                			d INT8,
                			e INT8,
                			f STRING)': pq: relation "target0" already exists
=== RUN   TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#02
I190815 05:25:42.447963 118401 sql/sqlbase/structured.go:1421  [n1,client=127.0.0.1:59046,user=root] publish: descID=95 (target0) version=2 mtime=2019-08-15 05:25:42.288690799 +0000 UTC
I190815 05:25:42.573918 118401 sql/event_log.go:130  [n1,client=127.0.0.1:59046,user=root] Event: "drop_database", target: 94, info: {DatabaseName:targetcols Statement:DROP DATABASE targetcols User:root DroppedSchemaObjects:[targetcols.public.target0]}
I190815 05:25:42.656776 118401 sql/sqlbase/structured.go:1421  [n1,client=127.0.0.1:59046,user=root,scExec] publish: descID=95 (target0) version=3 mtime=2019-08-15 05:25:42.650435363 +0000 UTC
        --- FAIL: TestImportIntoCSV/target-cols-with-default-pk/1,5,e,7,12,teststr#02 (0.02s)
            import_stmt_test.go:1585: error executing 'CREATE TABLE target0 (a INT8,
                			b INT8,
                			c STRING,
                			d INT8,
                			e INT8,
                			f STRING)': pq: relation "target0" already exists
=== RUN   TestImportIntoCSV/target-cols-with-default-pk
I190815 05:25:41.664494 118401 sql/event_log.go:130  [n1,client=127.0.0.1:59046,user=root] Event: "create_database", target: 94, info: {DatabaseName:targetcols Statement:CREATE DATABASE targetcols User:root}
    --- FAIL: TestImportIntoCSV/target-cols-with-default-pk (1.35s)

```

--- a/pkg/ccl/importccl/import_stmt.go
+++ b/pkg/ccl/importccl/import_stmt.go
@@ -761,18 +761,26 @@ func prepareExistingTableDescForIngestion(
 func (r *importResumer) prepareTableDescsForIngestion(
 	ctx context.Context, p sql.PlanHookState, details jobspb.ImportDetails,
 ) error {
-	importDetails := details
-	var hasExistingTables bool
 	err := p.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
-		var err error
+		importDetails := details
+		importDetails.Tables = make([]jobspb.ImportDetails_Table, len(details.Tables))
+
 		newTableDescToIdx := make(map[*sqlbase.TableDescriptor]int, len(importDetails.Tables))
+		var hasExistingTables bool
+		var err error
 		var newTableDescs []jobspb.ImportDetails_Table
-		for i, table := range importDetails.Tables {
+		var desc *sqlbase.TableDescriptor
+		for i, table := range details.Tables {
 			if !table.IsNew {
-				importDetails.Tables[i].Desc, err = prepareExistingTableDescForIngestion(ctx, txn, table.Desc, p)
+				desc, err = prepareExistingTableDescForIngestion(ctx, txn, table.Desc, p)
 				if err != nil {
 					return err
 				}
+				importDetails.Tables[i] = jobspb.ImportDetails_Table{Desc: desc, Name: table.Name,
+					SeqVal:     table.SeqVal,
+					IsNew:      table.IsNew,
+					TargetCols: table.TargetCols}
+
 				hasExistingTables = true
 			} else {
 				newTableDescToIdx[table.Desc] = i
@@ -793,8 +801,12 @@ func (r *importResumer) prepareTableDescsForIngestion(
 			if err != nil {
 				return err
 			}
-			for _, v := range res {
-				importDetails.Tables[newTableDescToIdx[v]].Desc = v
+			for i, table := range res {
+				importDetails.Tables[i] = jobspb.ImportDetails_Table{Desc: table,
+					Name:       details.Tables[i].Name,
+					SeqVal:     details.Tables[i].SeqVal,
+					IsNew:      details.Tables[i].IsNew,
+					TargetCols: details.Tables[i].TargetCols}
 			}
 		}
 
@@ -811,9 +823,9 @@ func (r *importResumer) prepareTableDescsForIngestion(
 
 		// Update the job once all descs have been prepared for ingestion.
 		err = r.job.WithTxn(txn).SetDetails(ctx, importDetails)
+
 		return err
 	})
-
 	return err
 }
 

